app/renderer/components/tabs/tab.js

```diff
@@ -128,12 +130,26 @@ class Tab extends React.Component {
     dnd.onDragOver(dragTypes.TAB, this.tabNode.getBoundingClientRect(), this.props.frameKey, this.draggingOverData, e)
   }
 
-  onMouseLeave () {
-    windowActions.setTabHoverState(this.props.frameKey, false)
+  onMouseLeave (e) {
+    clearTimeout(this.mouseTimeout)
+    // mouseleave will keep the previewMode
+    // as long as the related target is another tab
+    windowActions.setTabHoverState(this.props.frameKey, false, hasTabAsRelatedTarget(e))
   }
 
   onMouseEnter (e) {
-    windowActions.setTabHoverState(this.props.frameKey, true)
+    // if mouse entered another tab, preview will happen
+    // as long as the previewMode is enabled
+    windowActions.setTabHoverState(this.props.frameKey, true, this.props.previewMode)
+  }
+
+  onMouseMove () {
+    // previewMode is only triggered if mouse is idle over a tab
+    // for a given amount of time based on timing defined in prefs->tabs
+    clearTimeout(this.mouseTimeout)
+    this.mouseTimeout = setTimeout(() => {
```

Comment:
@bridiver I had a hard time moving the delayed state update to frameStateUtil while trying to keep using setTabHoverState (without adding extra actions). 

afaict there's no way to delay state updates without calling an action inside the utils file? Even if so I guess per our convo we are trying to avoid that. If this is not unacceptable I would prefer to leave it as-is.